app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -38,17 +39,24 @@ def send_authentication_notification
         @notification_enqueued = true
       end
 
+      def send_feed_customization_notification
+        return if user_is_following_tags? || received_notification?(customize_feed_broadcast) || user.created_at > 3.days.ago
+
+        Notification.send_welcome_notification(user.id, customize_feed_broadcast.id)
+      end
+
       def send_ux_customization_notification
         return if received_notification?(customize_ux_broadcast) || user.created_at > 5.days.ago
 
         Notification.send_welcome_notification(user.id, customize_ux_broadcast.id)
         @notification_enqueued = true
       end
 
-      def send_feed_customization_notification
-        return if user_is_following_tags? || received_notification?(customize_feed_broadcast) || user.created_at > 3.days.ago
+      def send_discuss_and_ask_notification
+        return if (asked_a_question && started_a_discussion) || received_notification?(discuss_and_ask_broadcast) || user.created_at > 6.days.ago
```

Comment:
```suggestion
        return if (asked_a_question? && started_a_discussion?) || received_notification?(discuss_and_ask_broadcast) || user.created_at > 6.days.ago
```
Since those two method sounds like a question and does return a boolean, I think it would be a good idea to add question marks to them.

Reply:
I agree with you in theory, but you can't add question marks to instance variables, which means we'd have something like this:

```
def asked_a_question?
    @asked_a_question ||= Article.user_published_with(user.id, "explainlikeimfive").any?
end
```

which, to me, is more confusing, because I'd expect the memoized variable to match the method name. Which is my justification for _not_ adding question marks to the methods 🤷‍♀